packages/material-ui/src/IconButton/IconButton.js

```diff
@@ -1,15 +1,55 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
-import { chainPropTypes } from '@material-ui/utils';
-import withStyles from '../styles/withStyles';
+import { chainPropTypes, deepmerge } from '@material-ui/utils';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
+import experimentalStyled from '../styles/experimentalStyled';
+import useThemeProps from '../styles/useThemeProps';
 import { alpha } from '../styles/colorManipulator';
 import ButtonBase from '../ButtonBase';
 import capitalize from '../utils/capitalize';
+import iconButtonClasses, { getIconButtonUtilityClass } from './iconButtonClasses';
 
-export const styles = (theme) => ({
-  /* Styles applied to the root element. */
-  root: {
+const overridesResolver = (props, styles) => {
+  const { styleProps } = props;
+
+  return deepmerge(styles.root || {}, {
+    ...(styleProps.disabled && styles.disabled),
+    ...(styleProps.color !== 'default' && styles[`color${capitalize(styleProps.color)}`]),
+    ...(styleProps.edge && styles[`edge${capitalize(styleProps.edge)}`]),
+    ...(styleProps.size === 'small' && styles.sizeSmall),
```

Comment:
```suggestion
    styles[`size${capitalize(styleProps.size)}`],
```

I don't see why we should not do this. cc @oliviertassinari

Reply:
I wanted to maintain consistency with the existing class overrides, but I guess it doesn't really make much of a difference, does it?  In fact, it probably is more beneficial to allow `size` to be `medium` since developers can create a class override for it!